src/clientHandler.ts

```diff
@@ -0,0 +1,252 @@
+	}
+
+	private newTerraformClient(location?: string): TerraformLanguageClient {
+		const cmd = this.pathToBinary;
+		const binaryName = cmd.split('/').pop();
+
+		const serverArgs: string[] = config('terraform').get('languageServer.args');
+		const experimentalFeatures = config('terraform-ls').get('experimentalFeatures');
+
+		let channelName = `${binaryName}`;
+		let id = `languageServer`
+		let name = `Language Server`
+		let wsFolder: vscode.WorkspaceFolder;
+		let rootModulePaths: string[];
+		let excludeModulePaths: string[];
+		let documentSelector: DocumentSelector;
+		let outputChannel: vscode.OutputChannel;
+		if (location) {
+			channelName = `${binaryName}: ${location}`;
+			id = `languageServer/${location}`
+			name = `Language Server: ${location}`
+			wsFolder = getWorkspaceFolder(location);
+			documentSelector = [
+				{ scheme: 'file', language: 'terraform', pattern: `${wsFolder.uri.fsPath}/**/*` },
+				{ scheme: 'file', language: 'terraform-vars', pattern: `${wsFolder.uri.fsPath}/**/*` }
+			]
+			rootModulePaths = config('terraform-ls', wsFolder).get('rootModules');
+			excludeModulePaths = config('terraform-ls', wsFolder).get('excludeRootModules');
+			outputChannel = vscode.window.createOutputChannel(channelName);
+			outputChannel.appendLine(`Launching language server: ${cmd} ${serverArgs.join(' ')} for folder: ${location}`);
+		} else {
+			documentSelector = [
+				{ scheme: 'file', language: 'terraform' },
+				{ scheme: 'file', language: 'terraform-vars' }
+			]
+			rootModulePaths = config('terraform-ls').get('rootModules');
+			excludeModulePaths = config('terraform-ls').get('excludeRootModules');
+			outputChannel = vscode.window.createOutputChannel(channelName);
+			outputChannel.appendLine(`Launching language server: ${cmd} ${serverArgs.join(' ')}`);
+		}
+
+		if (rootModulePaths.length > 0 && excludeModulePaths.length > 0) {
+			throw new Error('Only one of rootModules and excludeRootModules can be set at the same time, please remove the conflicting config and reload');
+		}
+
+		const commandPrefix = this.shortUid.seq();
+		let initializationOptions = { commandPrefix, experimentalFeatures };
+		if (rootModulePaths.length > 0) {
+			initializationOptions = Object.assign(initializationOptions, { rootModulePaths });
+		}
+		if (excludeModulePaths.length > 0) {
+			initializationOptions = Object.assign(initializationOptions, { excludeModulePaths });
+		}
+
+		const executable: Executable = {
+			command: cmd,
+			args: serverArgs,
+			options: {}
+		};
+		const serverOptions: ServerOptions = {
+			run: executable,
+			debug: executable
+		};
+		const clientOptions: LanguageClientOptions = {
+			documentSelector: documentSelector,
+			workspaceFolder: wsFolder,
+			initializationOptions: initializationOptions,
+			initializationFailedHandler: (error) => {
+				this.reporter.sendTelemetryException(error);
+				return false;
+			},
+			outputChannel: outputChannel,
+			revealOutputChannelOn: 4 // hide always
```

Comment:
There seems to be `RevealOutputChannelOn.Never` for that ([example](https://github.com/microsoft/vscode-extension-samples/blob/main/lsp-user-input-sample/client/src/extension.ts#L21)). ([Source](https://github.com/microsoft/vscode-languageserver-node/blob/0cb3812e7d540ef3a904e96df795bc37a21de9b0/client/src/common/client.ts#L358))

Reply:
Good catch!